hibernate-core/src/main/java/org/hibernate/query/hql/internal/SemanticQueryBuilder.java

```diff
@@ -1498,12 +1500,24 @@ public SqmDynamicInstantiation<?> visitInstantiation(HqlParser.InstantiationCont
 		}
 
 		if ( !dynamicInstantiation.checkInstantiation( creationContext.getTypeConfiguration() ) ) {
-			final String typeName = dynamicInstantiation.getJavaType().getSimpleName();
+			final Class<?> type = dynamicInstantiation.getJavaType();
 			if ( dynamicInstantiation.isFullyAliased() ) {
-				throw new SemanticException( "Missing constructor or attributes for injection into type '" + typeName + "'", query );
+				throw new SemanticException("Missing constructor or attributes for injection into type [" + type + "]." +
```

Comment:
Please back out this change and the other one like it. 

- we don't need punctuation at the end of exception messages, that's quite nonstandard, and
- I have been gradually migrating to the use of single quotes for quoting the names of program elements, since square brackets have more visual weight than is needed for this purpose.

Reply:
Thanks for reminding me, I have finished the modification